### ISSUE TITLE
panic in process_wrapper

### ISSUE THREAD

**Issue by UebelAndre** [bazelbuild/rules_rust#1209]
Earlier today I ran into the following error when just running `bazel test //...` on https://github.com/bazelbuild/rules_rust/commit/26e01c838679c8c0c52f4a82a7c973a3f5713960.
```
ERROR: /Users/user/Code/rules_rust/tools/rust_analyzer/BUILD.bazel:4:12: Clippy tools/rust_analyzer/gen_rust_project.clippy.ok failed: (Exit 101): process_wrapper failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/util/process_wrapper/process_wrapper --arg-file ... (remaining 82 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', util/process_wrapper/main.rs:78:24
```

The best repo I'm able to come up with is by running the following command
```shell
bazel clean && bazel test //tools/rust_analyzer:gen_rust_project_clippy --noremote_accept_cached
```

I'm running Bazel 5.0.0 on my Mac.

**Comment by gigaroby** [comment]
Hey @UebelAndre, mind trying the PR I just sent to fix (or at least get more debug info) for this?

**Comment by UebelAndre** [comment]
With the changes in https://github.com/bazelbuild/rules_rust/pull/1214 I got the following
```
user@Users-MacBook-Pro rules_rust % bazel clean --expunge && bazel test //tools/rust_analyzer:gen_rust_project_clippy --noremote_accept_cached
INFO: Invocation ID: 30d1a7b0-aa83-4cca-bd88-13e938a67742
INFO: Starting clean.
Starting local Bazel server and connecting to it...
INFO: Invocation ID: 888bd5a6-97ba-442c-9fbe-5aa740d2a48a
INFO: Analyzed target //tools/rust_analyzer:gen_rust_project_clippy (103 packages loaded, 3401 targets configured).
INFO: Found 1 target and 0 test targets...
ERROR: /Users/user/Code/rules_rust/tools/rust_analyzer/BUILD.bazel:4:12: Clippy tools/rust_analyzer/gen_rust_project.clippy.ok failed: (Exit 1): process_wrapper failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/util/process_wrapper/process_wrapper --arg-file ... (remaining 82 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox
process wrapper error: subprocess does not have an exit code. core_dumped: false, signal: Some(6), stopped_signal: None
Target //tools/rust_analyzer:gen_rust_project_clippy failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 66.955s, Critical Path: 36.16s
INFO: 183 processes: 117 internal, 66 darwin-sandbox.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
```

**Comment by gigaroby** [comment]
I can't repro this on linux, will try on my osx machine soon.

**Comment by illicitonion** [comment]
I reprod on macOS the first time I ran it, but can't repro again, no matter how much cleaning I do...

**Comment by gigaroby** [comment]
@illicitonion is that with or without the changes in #1214?

**Comment by illicitonion** [comment]
Without - I tried reproing with but also didn't get a failure.

**Comment by UebelAndre** [comment]
I seem to get it pretty consistently both with and without the changes. @illicitonion are you running the exact command I posted?
```shell
bazel clean --expunge && bazel test //tools/rust_analyzer:gen_rust_project_clippy --noremote_accept_cached
```
For some reason the expunge does seem to play a factor here

**Comment by illicitonion** [comment]
Yep, exact command, can't repro after the first attempt... Very weird!

**Comment by scentini** [comment]
I can't repro at HEAD with Bazel 5.1.0, macOS. @UebelAndre could you provide the failing command line?

**Comment by UebelAndre** [comment]
@scentini Hey, is this what you want?
https://github.com/bazelbuild/rules_rust/issues/1209#issuecomment-1076548442

**Comment by scentini** [comment]
Nah, i'd like to compare the arguments of the command that failed with mine.
`bazel clean --expunge && bazel test //tools/rust_analyzer:gen_rust_project_clippy -s --noremote_accept_cached` should give you what I want, or alternatively `bazel aqery //tools/rust_analyzer:gen_rust_project_clippy`.

**Comment by UebelAndre** [comment]
> Nah, i'd like to compare the arguments of the command that failed with mine. `bazel clean --expunge && bazel test //tools/rust_analyzer:gen_rust_project_clippy -s --noremote_accept_cached` should give you what I want, or alternatively `bazel aqery //tools/rust_analyzer:gen_rust_project_clippy`.

Sorry for the delay, my GitHub notifications have been weird and not actually notifying me of activity. The `aquery` gives nothing because the clippy build targets don't have actions and I'm now not able to reproduce the issue using the command 😞. Sometime this week I'll tinker some more to find a better repro.

**Comment by UebelAndre** [comment]
Closing this out since the issue appears to be resolved post Rust 1.60
